cloud-gateway-service/src/main/java/org/zowe/apiml/cloudgatewayservice/attls/AttlsHttpHandler.java

```diff
@@ -0,0 +1,139 @@
+import org.springframework.stereotype.Component;
+import org.springframework.web.server.adapter.DefaultServerWebExchange;
+import org.springframework.web.server.i18n.LocaleContextResolver;
+import org.springframework.web.server.session.DefaultWebSessionManager;
+import org.springframework.web.server.session.WebSessionManager;
+import org.zowe.apiml.message.core.Message;
+import org.zowe.apiml.message.core.MessageService;
+import org.zowe.commons.attls.*;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+@Component
+@RequiredArgsConstructor
+@ConditionalOnProperty("server.attls.enabled")
+public class AttlsHttpHandler implements BeanPostProcessor {
+
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    private final MessageService messageService;
+    private final LocaleContextResolver localeContextResolver;
+
+    private final WebSessionManager sessionManager = new DefaultWebSessionManager();
+    private final ServerCodecConfigurer serverCodecConfigurer = ServerCodecConfigurer.create();
+
+    private Mono<Void> writeError(ServerHttpRequest request, ServerHttpResponse response, String message) {
+        var serverWebExchange = new DefaultServerWebExchange(request, response, sessionManager, serverCodecConfigurer, localeContextResolver);
+        response.setRawStatusCode(500);
+        response.getHeaders().add(HttpHeaders.CONTENT_TYPE, APPLICATION_JSON_VALUE);
+        DataBuffer buffer = serverWebExchange.getResponse().bufferFactory().wrap(message.getBytes(StandardCharsets.UTF_8));
+        return serverWebExchange.getResponse().writeWith(Flux.just(buffer));
+    }
+
+    private String getMessage(String key) {
+        Message message = messageService.createMessage(key);
+        try {
+            return objectMapper.writeValueAsString(message.mapToView());
+        } catch (JsonProcessingException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    Mono<Void> internalError(ServerHttpRequest request, ServerHttpResponse response) {
+        return writeError(request, response, getMessage("org.zowe.apiml.gateway.internalServerError"));
+    }
+
+    Mono<Void> unsecureError(ServerHttpRequest request, ServerHttpResponse response) {
+        return writeError(request, response, getMessage("org.zowe.apiml.gateway.security.attls.notSecure"));
+    }
+
+    ServerHttpRequest updateCertificate(ServerHttpRequest request, HttpServletRequest nativeRequest, byte[] certificate) throws CertificateException {
+        if (ArrayUtils.isEmpty(certificate)) {
+            return request;
+        }
+
+        StringBuilder sb = new StringBuilder();
```

Comment:
since everything else is var as well
```suggestion
        var sb = new StringBuilder();
```